src/htm/algorithms/Connections.cpp

```diff
@@ -391,11 +391,10 @@ void Connections::reset()
   currentUpdates_.clear();
 }
 
-void Connections::computeActivity(
-    vector<SynapseIdx> &numActiveConnectedSynapsesForSegment,
+vector<SynapseIdx> Connections::computeActivity(
```

Comment:
please review if I'm passing reference to `vector&` correctly everywhere possible